pandas/core/tools/times.py

```diff
@@ -134,7 +134,18 @@ def _convert_listlike(arg, format):
 
 
 def _guess_time_format_for_array(arr):
-    # Try to guess the format based on the first non-NaN element
+    """
+    Try to guess the format based on the first non-NaN element
+    
+    Parameters
+    ----------
+    arr : string in unkown time format
```

Comment:
Hmm, why do you think this parameter is a string?

Reply:
I was looking at the parsing relation with the unknown format of the argument being passed which I think could be assumed to be a grouping of special characters? Would there be a better way to say it/a better way to understand it?

Reply:
The best way to understand it would be to find a `pandas` method that hits this function, then use a debugger to follow a simple example through and figure out how it gets here / what different inputs might be. The reason I'd be doubtful of it being a string is that the first line of the function is `notna(arr).nonzero()[0]` which would not work with `arr` as a `str`

Reply:
good call and thank you! I tracked it back to usage and found the input criteria applied to the variable